### PR TITLE
Migrate assertions to FluentAssertions

### PR DESCRIPTION
**Pull request by felixeu31**

### CODE REVIEW

File: Budgets/Budgets.Domain.UnitTests/Budgets.Domain.UnitTests.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="FluentAssertions" Version="6.7.0" />
```
Comment (tortosa) [inline Budgets/Budgets.Domain.UnitTests/Budgets.Domain.UnitTests.csproj]: Would be better in Budgets.Tests.Common so can be used in all test projects

File: Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs [hunk 2]
```diff
@@ -56,7 +57,7 @@ public void BudgetCategoryShouldHaveAssignedMoneyAtMonthYear()
                 .WithMoneyAssigned(monthYear, expectedMoneyAssigned)
                 .Build();
 
-            Assert.Equal(expectedMoneyAssigned, budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount));
+            budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount).Should().Be((decimal)expectedMoneyAssigned);//�Assert.Equal(expectedMoneyAssigned, budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount));?
```
Comment (tortosa) [inline Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: explicit decimal cast in expectedMoneyAssigned not required

Comment (tortosa) [inline Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: Remove comment

**Review by tortosa — CHANGES_REQUESTED** [review]

File: Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs (hunk 2 above)
Comment (felixeu31) [reply Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: Ok

File: Budgets/Budgets.Domain.UnitTests/Budgets.Domain.UnitTests.csproj (hunk 1 above)
Comment (felixeu31) [reply Budgets/Budgets.Domain.UnitTests/Budgets.Domain.UnitTests.csproj]: ok. I did not know that including the package only in a dependant referenced project would work

File: Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs (hunk 2 above)
Comment (felixeu31) [reply Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: Previous assert was performing an implicit cast from decimal to Money. In this case the asserted member defines the type in subsequent assertations, which is decimal as the Sum operation return that type. 

Would it be better to perform a explicit cast to Money for the Sum operation result? Or look the way of ensuring a sum operation which returns Money?

Comment (tortosa) [reply Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: Value is Money, but Amount is decimal

image

Comment (tortosa) [reply Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: In the previous version I had the explicit cast from Money, until I found Amount

Comment (felixeu31) [reply Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs]: I have a solution that could work. It is about an extension of Linq.Sum so it is possible to aggregate Money types. I followed: https://stackoverflow.com/questions/4132820/can-you-overload-sum-to-add-custom-types. Working locally, we can review it together

**Review by tortosa — APPROVED** [review]

**Pull request merged**